File: Homeworks/Homework1/Tire_Kutay.rb
# Associative Array Initialization

rubyDictionary = {"Assyria" => "Nineveh", "Babylonia" => "Babylon", "Sumeria" => "Ur", "Persia" => "Susa"}

# Getting a value

puts "Getting the value of 'Sumeria'"
val1 = rubyDictionary['Sumeria']
print "The value of 'Sumeria' is: ", val1
puts nil, nil 

# Adding a new element

puts "Adding a new element 'Egypt': "
rubyDictionary['Egypt'] = 'Thebes'
puts rubyDictionary

# Removing an element

puts nil, "Removing the element 'Persia': "
removedElement = rubyDictionary.delete('Persia')
if (removedElement)
    puts "'Persia' is successfully removed."
    puts rubyDictionary
else
    puts "'Persia' does not exist"
end

# Modifying an element

puts nil, "Modifying the element 'Assyria': "
rubyDictionary['Assyria'] = "Assur"
print(rubyDictionary)

# Searching for the existence of a key

puts nil,"Searching for 'Babylonia' key:"
foundBabylonia = false
rubyDictionary.each { |key, value| 
  
    if key == 'Babylonia'
        foundBabylonia = true
        break
    end
}

if (foundBabylonia)
  puts "'Babylonia' key is found."
else 
  puts "'Babylonia' key was not found."
end

# The key is not found

puts nil,"Searching for 'Rome' key:"
foundRome = false
rubyDictionary.each { |key, value| 
  
    if key == 'Rome'
        foundRome = true
        break
    end
}

if (foundRome)
  puts "'Rome' key is found."
else 
  puts "'Rome' key was not found."
end

# Searching for the existence of a value

puts nil, "Searching for 'Thebes' value"
foundThebes = false
rubyDictionary.each { |key, value| 
  
    if value == 'Thebes'
        foundThebes = true
        break
    end
}

if (foundThebes)
  puts "'Thebes' value is successfully found."
else 
  puts "'Thebes' was not found as a value."
end

# The value is not found

puts nil, "Searching for 'Athens' value"
foundAthens = false
rubyDictionary.each { |key, value| 
  
    if value == 'Athens'
        foundAthens = true
        break
    end
}

if (foundAthens)
  puts "'Athens' value is successfully found."
else 
  puts "'Athens' was not found as a value."
end

# Function declaration of foo() and printing key-value 

puts nil, "Looping through the array: "

def foo(key, value)
  puts "#{key}: #{value} "
end 

rubyDictionary.each { |key, value| 
       foo(key,value) 
}

File: Homeworks/Homework1/Tire_Kutay.rs
use std::collections::HashMap;
fn main() {
   
    let mut rust_dictionary = HashMap::new();
    rust_dictionary.insert("Assyria", "Nineveh");
    rust_dictionary.insert("Babylonia", "Babylon");
    rust_dictionary.insert("Sumeria", "Ur");
    rust_dictionary.insert("Persia", "Susa");
    
    // Getting a value
    
    println!("Getting the value of 'Sumeria': ");
    let value1 = rust_dictionary.get("Sumeria");
    match value1 {
        Some(value1) => println!("The value of 'Sumeria' is: {:?}", value1),
        None => println!("The value of 'Sumeria' does not exist."),
    }
    println!();
    
    // Adding a new element
    
    println!("Adding a new element 'Egypt': ");
    rust_dictionary.insert("Egypt", "Thebes");
    println!("{:?}", rust_dictionary);
    
    // Removing an element
    
    println!();
    println!("Removing the element 'Persia': ");
    rust_dictionary.remove("Persia");
    println!("{:?}", rust_dictionary);
    
    // Modifying an element
    
    println!();
    println!("Modifying the element 'Assyria': ");
    *rust_dictionary.get_mut("Assyria").unwrap() = "Assur";
    println!("{:?}", rust_dictionary);
    
    // Searching for the existence of a key
    
    println!();
    println!("Searching for 'Babylonia' key:");
    if rust_dictionary.contains_key("Babylonia"){
        println!("'Babylonia' key is found.");
    } 
    else{
        println!("'Babylonia' key was not found.");
    }
    
    println!();    
    println!("Searching for 'Rome' key:");
    if rust_dictionary.contains_key("Rome"){
        println!("'Rome' key is found.");
    } 
    else{
        println!("'Rome' key was not found.");
    }
    
    // Searching for the existence of a value
    
    println!();
    println!("Searching for 'Thebes' value");
    let mut found_thebes = false;
    for value in rust_dictionary.values() {
        if value == &"Thebes"{
            found_thebes = true;
            break;
        }
    }
    if found_thebes{
        println!( "'Thebes' value is successfully found.");
    } 
    else {
        println!("'Thebes' was not found as a value.");
    }
        
    println!();    
    println!("Searching for 'Athens' value");
    let mut found_athens = false;
    for value in rust_dictionary.values() {
        if value == &"Athens"{
            found_athens = true;
            break;
        }
    }
    if found_athens{
        println!( "'Athens' value is successfully found.");
    } 
    else {
        println!("'Athens' was not found as a value.");
    }
    
    // Function declaration of foo() and printing key-value
    
    println!();
    println!("Looping through the array: ");
    
    fn foo(key: &&str, value: &&str){
        println!("{}: {}", key, value);
    }
    for (key, value) in &rust_dictionary{
        foo(key, value);
    }
    
   
}